client/go/outline/client.go

```diff
@@ -124,5 +135,38 @@ func NewClientWithBaseDialers(clientConfigText string, tcpDialer transport.Strea
 		}
 	}
 
-	return &Client{sd: transportPair.StreamDialer, pl: transportPair.PacketListener}, nil
+	client := &Client{sd: transportPair.StreamDialer, pl: transportPair.PacketListener}
+	if clientConfig.Reporter != nil {
+		httpClient := &http.Client{
+			Transport: &http.Transport{
+				DialContext: func(ctx context.Context, network, addr string) (net.Conn, error) {
+					if strings.HasPrefix(network, "tcp") {
+						return client.DialStream(ctx, addr)
+					} else {
+						return nil, fmt.Errorf("protocol not supported: %v", network)
+					}
+				},
+			},
+		}
+		reporter, err := NewReporterParser(httpClient).Parse(context.Background(), clientConfig.Reporter)
+		if err != nil {
+			return nil, &platerrors.PlatformError{
+				Code:    platerrors.InvalidConfig,
+				Message: "invalid reporter config",
+				Cause:   platerrors.ToPlatformError(err),
+			}
+		}
+		client.reporter = reporter
+	}
+
+	return client, nil
+}
+
+func NewReporterParser(httpClient *http.Client) *configyaml.TypeParser[reporting.Reporter] {
```

Comment:
*nit* Shall we separate this out from `client.go`? And move it to another file?

Reply:
Maybe. For now I think it's easier to keep it here, because it's going to change. We can move it later.